smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -292,21 +378,42 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         );
     }
 
-    function _isManagedToken(address tokenAddress) private returns(bool) {
+    /**
+     * @dev Verifies if `tokenAddress` is a known token
+     * @param tokenAddress The address of the token
+     * @return Boolean: is `tokenAddress` a known token?
+     */
+    function _isManagedToken(address tokenAddress) private view returns(bool) {
         return sourceAddressToDestinationAddress[tokenAddress] != address(0);
     }
 
-    function verifyNetworkDescriptor(uint256 _networkDescriptor) internal returns(bool) {
+    /**
+     * @dev Verifies if `_networkDescriptor` matches this contract's networkDescriptor
+     * @param _networkDescriptor Unique identifier of the network
+     * @return Boolean: is `_networkDescriptor` what we expected?
+     */
+    function _verifyNetworkDescriptor(uint256 _networkDescriptor) private view returns(bool) {
         return _networkDescriptor == networkDescriptor;
     }
 
+    /**
+     * @dev Deploys a new BridgeToken, delegating this responsibility to BridgeBank
+     * @param symbol The symbol of the token
+     * @param name The name of the token
+     * @param sourceChainTokenAddress Address of the token on its original chain
+     * @param decimals The number of decimals this token has
+     * @param _networkDescriptor Unique identifier of the network
+     * @param cosmosDenom The token's Cosmos denom
+     * @return tokenAddress : The address of the newly deployed BridgeToken
+     */
     function _createNewBridgeToken(
-        string calldata symbol,
-        string calldata name,
+        string memory symbol,
```

Comment:
If we can make this into calldata, we should.

Let's do some gas benchmarking here and see how much things cost using memory.

Reply:
It costs 319 Gas for `symbol`, 319 Gas for `name`, and 587 Gas for `cosmosDenom`.
We can only change one of them to calldata, so I chose `cosmosDenom`.